core/src/main/java/org/wso2/msf4j/internal/MicroservicesRegistry.java

```diff
@@ -54,8 +53,19 @@
     private SessionManager sessionManager = new DefaultSessionManager();
 
     public MicroservicesRegistry() {
-        // Deploy the Swagger definition service which will return the Swagger definition.
-        services.add(new SwaggerDefinitionService(this));
+        /* If we can find the SwaggerDefinitionService, Deploy the Swagger definition service which will return the
+         Swagger definition.*/
+        try {
+            Class swaggerDefinitionServiceClass = Class.forName("org.wso2.msf4j.swagger.SwaggerDefinitionService");
+            services.add(swaggerDefinitionServiceClass.getConstructor(MicroservicesRegistry.class).newInstance(this));
+        } catch (ClassNotFoundException e) {
+            log.info("SwaggerDefinitionService can't be found in classpath.");
+        } catch (NoSuchMethodException | IllegalAccessException | InstantiationException
+                | InvocationTargetException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Couldn't add the SwaggerDefinitionService.", e);
+            }
+        }
```

Comment:
Will this work in OSGi mode? Is the new swagger jar and OSGi bundle and does it publish the Swagger OSGi service?